nbdime/webapp/nbdimeserver.py

```diff
@@ -101,181 +103,203 @@ def get_notebook_argument(self, argname):
+                args['base'] = base.name
             if isinstance(remote, string_types):
-                args["remote"] = remote
+                args['remote'] = remote
             else:
-                args["remote"] = remote.name
+                args['remote'] = remote.name
         else:
-            args["base"] = self.get_argument("base", "")
-            args["remote"] = self.get_argument("remote", "")
-        self.render("difftool.html", config_data=args, base_url=self.base_url)
+            args['base'] = self.get_argument('base', '')
+            args['remote'] = self.get_argument('remote', '')
+        self.write(self.render_template('difftool.html',
+                    config_data=args,
+                    mathjax_url=self.mathjax_url,
+                    mathjax_config=self.mathjax_config,
+                   ))
 
 
 class MainMergeHandler(NbdimeApiHandler):
     def get(self):
         args = self.base_args()
-        args["base"] = self.get_argument("base", "")
-        args["local"] = self.get_argument("local", "")
-        args["remote"] = self.get_argument("remote", "")
-        self.render("merge.html", config_data=args, base_url=self.base_url)
+        args['base'] = self.get_argument('base', '')
+        args['local'] = self.get_argument('local', '')
+        args['remote'] = self.get_argument('remote', '')
+        self.write(self.render_template('merge.html',
+                    config_data=args,
+                    mathjax_url=self.mathjax_url,
+                    mathjax_config=self.mathjax_config,
+                   ))
 
 
 class MainMergetoolHandler(NbdimeApiHandler):
     def get(self):
         args = self.base_args()
-        if "mergetool_args" in self.params:
-            args["base"] = self.params["mergetool_args"]["base"]
-            args["local"] = self.params["mergetool_args"]["local"]
-            args["remote"] = self.params["mergetool_args"]["remote"]
+        if 'mergetool_args' in self.params:
+            args['base'] = self.params['mergetool_args']['base']
+            args['local'] = self.params['mergetool_args']['local']
+            args['remote'] = self.params['mergetool_args']['remote']
         else:
-            args["base"] = self.get_argument("base", "")
-            args["local"] = self.get_argument("local", "")
-            args["remote"] = self.get_argument("remote", "")
-        self.render("mergetool.html", config_data=args, base_url=self.base_url)
+            args['base'] = self.get_argument('base', '')
+            args['local'] = self.get_argument('local', '')
+            args['remote'] = self.get_argument('remote', '')
+        self.write(self.render_template('mergetool.html',
+                    config_data=args,
+                    mathjax_url=self.mathjax_url,
+                    mathjax_config=self.mathjax_config,
+                   ))
 
 
-class ApiDiffHandler(NbdimeApiHandler):
+class ApiDiffHandler(NbdimeApiHandler, APIHandler):
```

Comment:
Is there a reason APIHandler is mixed in here as opposed to the base `NbdimeApiHandler`?

Reply:
Renamed `NbdimeApiHandler` -> `NbdimeHandler` to clear up the confusion.